src/bp/nlu-core/predict-pipeline.ts

```diff
@@ -46,9 +44,8 @@ interface InitialStep {
   languageCode: string
 }
 type PredictStep = InitialStep & { utterance: Utterance }
-type OutOfScopeStep = PredictStep & { oos_predictions: _.Dictionary<number> }
-type ContextStep = OutOfScopeStep & { ctx_predictions: MLToolkit.SVM.Prediction[] }
-type IntentStep = ContextStep & { intent_predictions: _.Dictionary<MLToolkit.SVM.Prediction[]> }
+type ContextStep = PredictStep & { ctx_predictions: MLToolkit.SVM.Prediction[] }
```

Comment:
this is probably in your plan but just in case it's not. You completely abstracted SVM away from text classification. Text classification task should have it's own prediction interface and dig svm away

Reply:
Yep, its in plans.